### PR TITLE
Add 802154-subg test

### PR DESCRIPTION
**Pull request by Ayush1325**
- Using beagleconnect freedom for testing subg overlay
- Add to zperf, echo_server and echo_client

### CODE REVIEW

**Comment by jadonk** [comment]
@Ayush1325 should the echo_server be enabled in the same commit?

**Comment by Ayush1325** [comment]
> @Ayush1325 should the echo_server be enabled in the same commit?

Probably not. They are different samples and have their own copies of the 802154-subg overlay. 

They can be part of the same PR  but I only added it to echo_client for now to check if upstream accepts this change since it does affect the CI.

**Review by rlubos — DISMISSED** [review]

**Review by jukkar — DISMISSED** [review]
Related to #79504, please use `EXTRA_CONF_FILE` instead.

**Review by jukkar — DISMISSED** [review]

**Review by rlubos — DISMISSED** [review]

**Comment by Ayush1325** [comment]
Seems I actually caught a bug. Need to find what is wrong here.

File: subsys/net/l2/ieee802154/ieee802154_shell.c [hunk 1]
```diff
@@ -99,7 +99,7 @@ static int cmd_ieee802154_associate(const struct shell *sh,
 
 	params = (struct ieee802154_req_params){0};
 	params.pan_id = atoi(argv[1]);
-	strncpy(ext_addr, argv[2], sizeof(ext_addr));
+	strncpy(ext_addr, argv[2], EXT_ADDR_STR_LEN);
```
Comment (jukkar) [inline subsys/net/l2/ieee802154/ieee802154_shell.c:102]: Why this change is needed, the sizeof() should have exactly the same value as the define?

Comment (Ayush1325) [reply subsys/net/l2/ieee802154/ieee802154_shell.c:102]: See https://github.com/zephyrproject-rtos/zephyr/blob/9f60075d14a283602a3580b6b97dd4686b53a28e/subsys/net/l2/ieee802154/ieee802154_shell.c#L25

The length of ext_addr is EXT_ADDR_STR_SIZE. 
EXT_ADDR_STR_LEN = EXT_ADDR_STR_SIZE - 1;

In line 103, we are using `strlen` which expects a NULL terminator. Thus we should only copy `sizeof(ext_addr) - 1` number of elements.

Comment (jukkar) [reply subsys/net/l2/ieee802154/ieee802154_shell.c:102]: Ah, my bad, I did not spot that there was two different defines, the other with SIZE and the other LEN. Sorry for the noise.

**Review by jukkar — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Review by ghost — APPROVED** [review]
Thanks!
